qiskit_experiments/randomized_benchmarking/rb_experiment.py

```diff
@@ -0,0 +1,131 @@
+from .rb_analysis import RBAnalysis
+
+
+class RBExperiment(BaseExperiment):
+    """RB Experiment class"""
+
+    # Analysis class for experiment
+    __analysis_class__ = RBAnalysis
+
+    def __init__(
+        self,
+        qubits: Union[int, Iterable[int]],
+        lengths: Iterable[int],
+        num_samples: int = 1,
+        seed: Optional[Union[int, Generator]] = None,
+        full_sampling: bool = False,
+    ):
+        """Standard randomized benchmarking experiment
+        Args:
+            qubits: the number of qubits or list of
+                    physical qubits for the experiment.
+            lengths: A list of RB sequences lengths.
+            num_samples: number of samples to generate for each
+                         sequence length
+            seed: Seed or generator object for random number
+                  generation. If None default_rng will be used.
+            full_sampling: If True all Cliffords are independently sampled for
+                           all lengths. If False for sample of lengths longer
+                           sequences are constructed by appending additional
+                           Clifford samples to shorter sequences.
+        """
+        if not isinstance(seed, Generator):
+            self._rng = default_rng(seed=seed)
+        else:
+            self._rng = seed
+        self._lengths = list(lengths)
+        self._num_samples = num_samples
+        self._full_sampling = full_sampling
+        super().__init__(qubits)
+
+    # pylint: disable = arguments-differ
+    def circuits(self, backend=None):
+        """Return a list of RB circuits.
+        Args:
+            backend (Backend): Optional, a backend object.
+        Returns:
+            List[QuantumCircuit]: A list of :class:`QuantumCircuit`s.
+        """
+        circuits = []
+        for _ in range(self._num_samples):
+            circuits += self._sample_circuits(self._lengths, seed=self._rng)
+        return circuits
+
+    def transpiled_circuits(self, backend=None, **kwargs):
+        """Return a list of transpiled RB circuits.
+        Args:
+            backend (Backend): Optional, a backend object to use as the
+                               argument for the :func:`qiskit.transpile`
+                               function.
+            kwargs: kwarg options for the :func:`qiskit.transpile` function.
+        Returns:
+            List[QuantumCircuit]: A list of :class:`QuantumCircuit`s.
+        Raises:
+            QiskitError: if an initial layout is specified in the
+                         kwarg options for transpilation. The initial
+                         layout must be generated from the experiment.
+        """
+        circuits = super().transpiled_circuits(backend=backend, **kwargs)
+        # TODO: Add functionality for gates per clifford which depends
+        # on the transpiled circuit gates
+        return circuits
+
+    def _sample_circuits(self, lengths, seed=None):
```

Comment:
Documentation is missing here and in `_generate_circuit`

Reply:
Thanks, linting was not done yet at this stage.